java/ql/lib/upgrades/ecfcf050952e54b1155fc89525db84af6ad34aaf/exprs.ql

```diff
@@ -0,0 +1,47 @@
+class Expr extends @expr {
+  string toString() { result = "expr" }
+}
+
+class LocalVariableDeclExpr extends @localvariabledeclexpr, Expr { }
+
+class InstanceOfExpr extends @instanceofexpr, Expr { }
+
+class Type extends @type {
+  string toString() { result = "type" }
+}
+
+class ExprParent extends @exprparent {
+  string toString() { result = "exprparent" }
+}
+
+// Initialisers of local variable declarations that occur as the 0th child of an instanceof expression should be reparented to be the 0th child of the instanceof itself,
+// while the LocalVariableDeclExpr, now without an initialiser, should become its 2nd child.
+// This implements a reorganisation of the representation of "o instanceof String s", which used to be InstanceOfExpr -0-> LocalVariableDeclExpr --init-> o
+//                                                                                                                                               \-name-> s
+// It is now InstanceOfExpr --0-> o
+//                          \-2-> LocalVariableDeclExpr -name-> s
+//
+// Other children are unaffected.
+ExprParent getParent(Expr e) { exprs(e, _, _, result, _) }
+
+predicate hasNewParent(Expr e, ExprParent newParent, int newIndex) {
+  if
+    getParent(e) instanceof LocalVariableDeclExpr and
+    getParent(getParent(e)) instanceof InstanceOfExpr
+  then (
+    newParent = getParent(getParent(e)) and newIndex = 0
+  ) else (
+    // Initialiser moves to hang directly off the instanceof expression
```

Comment:
Looks like this and the next comment dropped a few lines down (likely due to autoformat). We should move them up a bit to make this less confusing.